rir/src/compiler/analysis/abstract_value.cpp

```diff
@@ -124,7 +125,12 @@ AbstractLoad AbstractREnvironmentHierarchy::get(Value* env, SEXP e) const {
         auto aenv = envs.at(env);
         if (!aenv.absent(e)) {
             const AbstractPirValue& res = aenv.get(e);
-            return AbstractLoad(env, res);
+            // UnboundValue has fall-through semantics which cause lookup to
+            // fall through.
+            if (res.maybeUnboundValue())
+                return AbstractLoad(env, AbstractPirValue::tainted());
+            if (!res.isUnboundValue())
```

Comment:
Can this happen? I do not see it.

Reply:
Currently probably not. But it does not hurt. e.g. we could in the future track `rm` and implement it by writing unboundValue...